### ISSUE TITLE
Subscription rollover issue

### ISSUE THREAD

**Issue by g2graman** [HabitRPG/habitica#4480]
User id:
9bc07270-4c5a-4410-8030-1d41e6e81aa1

Group ID: 4889c225-b5ed-41b1-92c2-bbc595521e62
Browser: Chrome
OS: Windows 8.1

I was gifted a 3 month subscription last month and expected my gem cap (etc.) to roll over at the start of this month and have given it a few days but it still hasn't happened yet.

**Comment by lefnire** [comment]
Strange... I've reset you, give that a go

**Comment by lefnire** [comment]
Got another mention of this via email, bumping the priority

**Comment by wildcate** [comment]
Same issue with my husband's account and mine - we both self-gifted a block subscription. There was a gem-buying issue with the gift subscriptions when they became first available, maybe the rollover problem is linked to that?

**Comment by lefnire** [comment]
TL;DR, it will be fixed on your next cron (tomorrow). Ping back if it's not reset tomorrow.

Long: Gifting wasn't setting `member.purchased.plan.dateUpdated` (if `!null`) meaning recipients didn't have a `dateUpdated` to check against for reseting gemCapExtra (`moment(undefined).format('MMYYYY') == today` at https://github.com/HabitRPG/habitrpg-shared/blob/54a9df4619e033fe9d590c6a69fe755200dff3f3/script/index.coffee#L1419).

**Comment by wildcate** [comment]
Ping! Reset has not happened for me yet (I tried refresh, hard refresh and logging out and in again). 
I also have an issue with the town crier, who appears but brings old news only - I can only get the new news by clicking on the "read old news", since the message is always one newsflash behind. (Probably not related, though.)

**Comment by Alys** [comment]
@wildcate Yes, that's not related. Whenever you see old news in the town crier window, do NOT dismiss that window. Instead just reload the page using your browser's refresh function. All that's happened is that your browser hasn't loaded the new content.

Please tell us your User ID (from the Settings -> API page). Also, has your husband's reset correctly? If not, tell us his user ID as well.

Back to @lefnire for the actual subscription issue.

**Comment by Alys** [comment]
@wildcate Please tell us your User ID (from the Settings -> API page). Also, has your husband's reset correctly? If not, tell us his user ID as well.

**Comment by wildcate** [comment]
Thanks @Alys - I had not thought of doing a refresh with the town crier!

My user ID is: d7ca2b94-7515-4f88-bed0-cb16160e4b5f
and that of my husband (who has not had a reset either): a7886d4b-e617-4bd4-8698-c2c5f6224b26

**Comment by Alys** [comment]
@wildcate Have the gems caps still not reset for you and your husband?

**Comment by wildcate** [comment]
Still no reset, no....

**Comment by Alys** [comment]
@wildcate I've manually reset the count of gems bought for both of you. Let me know if it hasn't worked. Next month, if you experience the same problem, be sure to let us know!

**Comment by wildcate** [comment]
Thank you! (By the way, the cap had re-set and the gem showed the reset number of gems available, but until after the first gem bought, it was still translucent.)

**Comment by emeraldmir** [comment]
My friend gifted me a three-month subscription in December, and I was able to buy my 30 gems with gold before December 31. But the gems didn't reset in January, and it's been a few weeks now. I have logged out and back in, as well as cleared my browser's cache or tried a different computer.
My User ID: e3fcddc8-495a-4d13-b616-36aeeebbe233
Safari, Mac OSX Yosemite.

**Comment by Alys** [comment]
investigating now

**Comment by Alys** [comment]
@emeraldmir I've reset your gem balance so you should be able to buy all of your monthly gems now. You can ignore the rest of this comment if you wish; it's just about fixing the bug for other users too.

@lefnire: I've also reset emeraldmir's purchased.plan.dateUpdated value to today, so that cron won't cause the balance to be reset again until next month.

The problem was this:
https://github.com/HabitRPG/habitrpg/blob/0b28f6df55f43792b8e4234f56473c86a921259d/migrations/20150107_plan_dateUpdated_null.js#L3
`datedUpdated` instead of `dateUpdated` (extra d before U)

I've updated that script locally and am about to test on a local copy of the database (I just have to get a recent copy running first which takes a while). Once that's done and successful, I'll commit the script and seek your permission to run it over the production database. That should fix this problem for all gift subscribers.

**Comment by Alys** [comment]
I've run the script over the production database after it was checked by lefnire. For the remaining people who did not have their gems balance reset, it will be reset after their next cron, even if they are resting in the inn (a mini-cron runs for sleeping accounts).
